src/sidebars/favourites/favourite-controller.js

```diff
@@ -129,25 +131,25 @@
 
                             // Repeat the check as in the mean time a stock for this favourite could have been added.
                             if (this.stocks.map((stock1) => { return stock1.code; }).indexOf(favourite) === -1) {
-                                var data = stock.data[0],
+                                var data = stock && stock.data && stock.data[0],
```

Comment:
In theory checking the success property should work for almost all cases.
I wrote this thinking there's a chance success data will be missing, although admittedly it'd show an empty code and message.
I felt this provided a bit more robustness, since a malformed success message would break our app.
I've added a stock check to the _addError to be on the safe side